bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/api/EcobeeApi.java

```diff
@@ -344,11 +346,13 @@ private boolean isSuccess(@Nullable AbstractResponseDTO response) {
                 createOAuthClientService();
             } else if (response.status.code == ECOBEE_TOKEN_EXPIRED) {
                 // Check isAuthorized again to see if we can get a valid token
-                logger.debug("API: Unable to complete API call because token is expired");
+                logger.info("API: Unable to complete API call because token is expired. Try to refresh the token...");
+                closeOAuthClientService();
+                createOAuthClientService();
                 if (isAuthorized()) {
                     return true;
                 } else {
-                    logger.debug("API: isAuthorized was NOT successful on second try");
+                    logger.info("API: isAuthorized was NOT successful on second try");
```

Comment:
I guess this is a final state? If so this should be warn or the Thing status should be set (i'm unable to determine this in this short review)

Reply:
Yes, you're probably right that the status should be set to OFFLINE.